contracts/walrus/sources/blob_manager.move

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) Walrus Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+/// Minimal blob-management interface using enum-based storage strategies.
+/// This version provides only essential functionality: new, register, and certify.
+module walrus::blobmanager;
+
+use sui::coin::Coin;
+use wal::wal::WAL;
+use walrus::{
+    blob_stash::{Self, BlobStash},
+    blob_storage::{Self, BlobStorage},
+    encoding,
+    storage_resource::Storage,
+    system::{Self, System}
+};
+
+// === Constants ===
+
+/// Minimum initial capacity for BlobManager (500MB in bytes).
+const MIN_INITIAL_CAPACITY: u64 = 500_000_000; // 500 MB.
+
+// === Error Codes ===
+
+/// The provided BlobManagerCap does not match the BlobManager.
+const EInvalidBlobManagerCap: u64 = 0;
+/// Initial storage capacity is below minimum requirement.
+const EInitialBlobManagerCapacityTooSmall: u64 = 3;
+/// The blob is already certified (cannot register or certify again).
+const EBlobAlreadyCertifiedInBlobManager: u64 = 4;
+/// The requested blob was not found in BlobManager.
+const EBlobNotRegisteredInBlobManager: u64 = 2;
+
+// === Main Structures ===
+
+/// The minimal blob-management interface.
+public struct BlobManager has key, store {
+    id: UID,
+    /// Storage management strategy.
+    storage: BlobStorage,
+    /// Blob stash strategy - handles both individual blobs and managed blobs.
+    blob_stash: BlobStash,
+}
+
+/// A capability which represents the authority to manage blobs in the BlobManager.
+public struct BlobManagerCap has key, store {
```

Comment:
How about adding a permission bitfield.
WRITE, DELETE, METADATA, ADMIN
Only Admin can create new ManagerCaps. This would allow more fine granular permissions. Give buildbots for example only permission to create new artifacts.